core/src/main/java/org/keycloak/Config.java

```diff
@@ -91,7 +91,7 @@ public String get(String key) {
         @Override
         public String get(String key, String defaultValue) {
             String v = System.getProperty(prefix + key, defaultValue);
-            return v != null && !v.isEmpty() ? v : null;
+            return v != null && !v.isEmpty() ? v : defaultValue;
```

Comment:
Please revert this change. Either this is a bug that should be treated separately, or a workaround that should be treated properly.

Reply:
Reverted